assets/js/collaborative-editor/stores/createWorkflowStore.ts

```diff
@@ -474,7 +474,12 @@ export const createWorkflowStore = () => {
     // Observer handles the rest: Y.Doc → immer → notify
   };
 
-  const addJob = (job: Partial<Session.Job>) => {
+  const addJob = (
+    job: Partial<Session.Job> & {
+      project_credential_id?: string | null;
+      keychain_credential_id?: string | null;
+    }
+  ) => {
```

Comment:
This isn't right, these attributes should be on Session.Job; i.e. they are missing from the type. Not sure why they were unioned on the arguments.

Also they shouldn't be optional.